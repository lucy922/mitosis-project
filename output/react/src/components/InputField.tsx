import * as React from "react";

export interface IinputFieldClasses {
  base: string;
}
export interface IinputFieldProps {
  type?: string;
  placeholder?: string;
  name?: string;
  onChange?: () => void;
  value?: string;
}

import classes from "./classes/InputField.classes";

export default function InputField(props: IinputFieldProps) {
  return (
    <input
      {...props}
      type={props.type}
      placeholder={props.placeholder}
      name={props.name}
      onChange={(event) => props.onChange}
      value={props.value}
      className={classes.base}
    />
  );
}
