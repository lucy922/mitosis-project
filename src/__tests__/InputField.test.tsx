import { screen, render } from "@testing-library/react";
import { fireEvent } from "@testing-library/react";
import InputField from "../../output/react/src/components/InputField";

describe("should render input field correctly", () => {
  it("email input field should accept email", () => {
    render(<InputField placeholder="Enter email.." />);
    const email =
      screen.getByPlaceholderText<HTMLInputElement>("Enter email..");
    fireEvent.change(email, { target: { value: "ofemlucy@gmail.com" } });
    expect(email.value).toBe("ofemlucy@gmail.com");
    expect(email).toBeInTheDocument();
    expect(email).toMatchSnapshot();
  });

  it("password input field should be type password", () => {
    render(<InputField placeholder="Enter password" type="password" />);
    const password =
      screen.getByPlaceholderText<HTMLInputElement>("Enter password");
    fireEvent.change(password, { target: { value: "1234567" } });
    expect(password.value).toBe("1234567");
    expect(password).toHaveAttribute("type", "password");
    expect(password).toBeInTheDocument();
    expect(password).toMatchSnapshot();
  });
});
