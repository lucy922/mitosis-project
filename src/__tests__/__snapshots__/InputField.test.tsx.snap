// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render input field correctly email input field should accept email 1`] = `
<input
  class="border-solid border-2 border-sky-300 outline-none w-96 rounded-md p-2"
  placeholder="Enter email.."
  value=""
/>
`;

exports[`should render input field correctly password input field should be type password 1`] = `
<input
  class="border-solid border-2 border-sky-300 outline-none w-96 rounded-md p-2"
  placeholder="Enter password"
  type="password"
  value=""
/>
`;
